src/backend/distributed/transaction/transaction_recovery.c

```diff
@@ -511,3 +511,37 @@ RecoverPreparedTransactionOnWorker(MultiConnection *connection, char *transactio
 
 	return true;
 }
+
+
+/*
+ * DeleteWorkerTransactions deletes the entries on pg_dist_transaction for a given
+ * worker node. It's implemented to be called at master_remove_node.
+ */
+void
+DeleteWorkerTransactions(WorkerNode *workerNode)
+{
+	bool indexOK = true;
+	int scanKeyCount = 1;
+	ScanKeyData scanKey[1];
+	int32 groupId = workerNode->groupId;
+	HeapTuple heapTuple = NULL;
+
+	Relation pgDistTransaction = table_open(DistTransactionRelationId(),
+											RowExclusiveLock);
+
+	ScanKeyInit(&scanKey[0], Anum_pg_dist_transaction_groupid,
+				BTEqualStrategyNumber, F_INT4EQ, Int32GetDatum(groupId));
+
+	SysScanDesc scanDescriptor = systable_beginscan(pgDistTransaction,
+													DistTransactionGroupIndexId(),
+													indexOK,
+													NULL, scanKeyCount, scanKey);
+
+	while (HeapTupleIsValid(heapTuple = systable_getnext(scanDescriptor)))
+	{
+		simple_heap_delete(pgDistTransaction, &heapTuple->t_self);
+	}
+
+	systable_endscan(scanDescriptor);
```

Comment:
Probably not very critical here, but we typically do the following when we modify tuples, please see the code for examples:
```C
	/* increment the counter so that next command won't see the row */
	CommandCounterIncrement();
```